### PR TITLE
EnItem00 OK

### PR DESCRIPTION
**Pull request by hensldm**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

Matches the three NON-MATCHINGs in EnItem00.

### CODE REVIEW

**Review by EllipticEllipsis — COMMENTED** [review]
So all those functions have different params types? What a pain.

**Comment by hensldm** [comment]
> So all those functions have different params types? What a pain.

My bad thought I had changed it. The first itemDrop can also have `s32` for its params.

File: src/code/z_en_item00.c [hunk 1]
```diff
@@ -881,59 +872,50 @@ EnItem00* Item_DropCollectible(GlobalContext* globalCtx, Vec3f* spawnPos, u32 pa
...
         if (param8000 == 0) {
-            params = func_800A7650(params & 0xFF);
+            params = func_800A7650(newParamFF);
         }
-        if (params != (u32)ITEM00_NO_DROP) {
-            spawnedActor = (EnItem00*)Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ITEM00, spawnPos->x,
-                                                  spawnPos->y, spawnPos->z, 0, 0, 0, params | param8000 | param7F00);
+        if ((s32)params != ITEM00_NO_DROP) {
+            spawnedActor = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ITEM00, spawnPos->x, spawnPos->y,
+                                       spawnPos->z, 0, 0, 0, (s32)params | param8000 | param7F00);
```
Comment (EllipticEllipsis) [inline src/code/z_en_item00.c:879]: These casts still needed?

File: src/code/z_en_item00.c [hunk 2]
```diff
@@ -945,10 +927,10 @@ Actor* Item_DropCollectible2(GlobalContext* globalCtx, Vec3f* spawnPos, u32 para
...
         }
     } else {
-        params = func_800A7650(params);
-        if (params != (u32)ITEM00_NO_DROP) {
+        params = func_800A7650(params & 0xFF);
+        if ((s32)params != ITEM00_NO_DROP) {
             spawnedActor = Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_ITEM00, spawnPos->x, spawnPos->y,
-                                       spawnPos->z, 0, 0, 0, params | param8000 | param7F00);
+                                       spawnPos->z, 0, 0, 0, (s32)params | param8000 | param7F00);
             if (spawnedActor != NULL) {
```
Comment (EllipticEllipsis) [inline src/code/z_en_item00.c:934]: And here?

File: src/code/z_en_item00.c (hunk 1 above)
Comment (hensldm) [reply src/code/z_en_item00.c]: the conditional no, the or does it need for regalloc reasons.

File: src/code/z_en_item00.c (hunk 2 above)
Comment (hensldm) [reply src/code/z_en_item00.c]: Same as above

**Review by AngheloAlf — APPROVED** [review]

**Review by EllipticEllipsis — APPROVED** [review]

**Comment by hensldm** [comment]
So it seems each of these functions actually do have different param types. When merging BigSlime, I couldn't get `EnBigslime_ApplyDamageEffectBigslime` to match without changing `Item_DropCollectible`'s params back to a `u32`.

File: src/code/z_en_item00.c [hunk 3]
```diff
@@ -1188,9 +1168,7 @@ s32 func_800A8150(s32 index) {
     return D_801AE194[index];
 }
 
-u8 D_801AE214[32] = {
-    0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-};
+u8 D_801AE214[32] = { 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
```
Comment (engineer124) [inline src/code/z_en_item00.c:1171]: I think I very slightly prefer this is it was previously with the comma, but it's not a big deal

**Review by engineer124 — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**